csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1211 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Apache.Arrow.Flight.Client;
+using Apache.Arrow.Flight.Server;
+using Apache.Arrow.Types;
+using Arrow.Flight.Protocol.Sql;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight.Sql.Client;
+
+public class FlightSqlClient
+{
+    private readonly FlightClient _client;
+
+    public FlightSqlClient(FlightClient client)
+    {
+        _client = client ?? throw new ArgumentNullException(nameof(client));
+    }
+
+    public static Transaction NoTransaction() => new(null);
```

Comment:
This should be a static readonly variable to avoid unneeded allocations

Reply:
done